src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1799,21 +1841,8 @@ private static EmitResult ToEmitResultAndFree(DiagnosticBag diagnostics, bool su
                         }
 
                         portablePdbStream = pdbStreamProvider.GetOrCreateStream(metadataDiagnostics);
-                        if (portablePdbStream == null)
-                        {
-                            Debug.Assert(metadataDiagnostics.HasAnyErrors());
-                            return null;
-                        }
-
-                        // When in deterministic mode, we need to seek and read the stream to compute a deterministic PDB ID.
-                        // If the underlying stream isn't readable and seekable, we need to use a temp stream.
-                        var retStream = portablePdbStream;
-                        if (!retStream.CanSeek || deterministic && !retStream.CanRead)
-                        {
-                            retStream = portablePdbTempStream = new MemoryStream();
-                        }
-
-                        return retStream;
```

Comment:
No, deleted. We don't need the stream to be seekable anymore since we calculate the hash in place.